cmake/extensions.cmake

```diff
@@ -389,6 +389,19 @@ macro(zephyr_library)
   zephyr_library_named(${lib_name})
 endmacro()
 
+# Checks if a Kconfig is enabled and then does the same operation as
+# zephyr_library. If it finds that the Kconfig is not set it will not
+# create a new library and will stop processing the rest of
+# the CMakeFile.txt
+macro(zephyr_library_ifdef config)
```

Comment:
Instead of 'config' should use the same naming convention as the other *_ifdef functions ('feature_toogle').